src/filters.js

```diff
@@ -1,16 +1,68 @@
 const createDOMPurify = require('dompurify');
 const { JSDOM } = require('jsdom');
 
+const ALLOWED_ATTRS_BY_TAG_FOR_HEAD_MARKUP = {
+  meta: ['name', 'content', 'property'],
+  link: ['rel', 'href'],
+  script: ['type'],
+};
+
 const makeBetterSafe = ({ markAsSafe }) => {
   const { window } = new JSDOM('');
   const DOMPurify = createDOMPurify(window);
 
-  return (value) => {
+  return (value, { isHeadMarkup = false } = {}) => {
     if (!value) {
-      return;
+      return undefined;
+    }
+
+    if (isHeadMarkup) {
+      const domPurifyOptions = {
+        WHOLE_DOCUMENT: true,
+        // This is needed because `WHOLE_DOCUMENT` is required to sanitize
+        // `meta` tags but DOMPurify will return the content inside
+        // `<html><head>...</head></html>` instead of "just" the meta tags.
+        RETURN_DOM: true,
+        // This is needed to allow property values like `og:xyz`.
+        ADD_URI_SAFE_ATTR: ['property'],
```

Comment:
Does this allow it but still sanitize the value?

Reply:
That's my understanding but I can add a test case to verify.

Reply:
@muffinresearch it doesn't sanitize the value. It marks `property` as safe like `id`, `name`, `xmlns`, etc. Those attributes are safe for `javascript:` like values according to [this](https://github.com/cure53/DOMPurify/blob/e1c19cf6407d782b666cb1d02a6af191f9cbc09e/src/purify.js#L307) and this solution is recommended by the author: https://github.com/cure53/DOMPurify/issues/524#issuecomment-798900327, so maybe it's fine for `property` as well?

Reply:
WDYT?

Reply:
If there's no way to abuse the allowed attribute then we probably shouldn't be too concerned. 

If quotes are included (in the value) are they escaped? Can you close the meta tag and inject other HTML (though presumably that would be sanitized independently?) 

Are there varying values for that attr? If not could the value be validated?

Reply:
> If quotes are included (in the value) are they escaped? Can you close the meta tag and inject other HTML (though presumably that would be sanitized independently?)

I did that in a test case, for the `content` attribute. I suppose that'd be the same for any other attribute.

> Are there varying values for that attr? If not could the value be validated?

I think we could only accept `og:*` values.

Reply:
If you can't break out of the attribute, then we probably don't need to worry too much - what do you think? Maybe adding a test for that specific to the `property` attribute would at least cover that since we're relying on it.

Reply:
added another test case